tests/test_matrix_utils.py

```
@@ -2262,6 +2375,525 @@ async def test_send_room_image():
+
+# E2EE Configuration Tests
+@patch("mmrelay.matrix_utils.os.makedirs")
+@patch("mmrelay.matrix_utils.os.listdir")
+@patch("mmrelay.matrix_utils.os.path.exists")
+@patch("builtins.open")
+@patch("mmrelay.matrix_utils.json.load")
+@patch("mmrelay.matrix_utils._create_ssl_context")
+@patch("mmrelay.matrix_utils.matrix_client", None)
+@patch("mmrelay.matrix_utils.AsyncClient")
+@patch("mmrelay.matrix_utils.logger")
+async def test_connect_matrix_restore_login_uses_none_device_id(
```

Ruff (0.14.5)

Redefinition of unused `test_connect_matrix_restore_login_uses_none_device_id` from line 2522

(F811)